modules/edit-modes/src/lib/measure-distance-mode.ts

```diff
@@ -1,137 +1,180 @@
-// @ts-ignore
 import turfDistance from '@turf/distance';
-// @ts-ignore
+import { FeatureCollection } from '../geojson-types';
+import { ClickEvent, PointerMoveEvent, ModeProps, GuideFeatureCollection, Tooltip } from '../types';
+import { getPickedEditHandle } from '../utils';
 import memoize from '../memoize';
-import {
-  ClickEvent,
-  PointerMoveEvent,
-  Tooltip,
-  ModeProps,
-  GuideFeatureCollection,
-  EditHandleFeature,
-} from '../types';
-import { FeatureCollection, Position } from '../geojson-types';
 import { GeoJsonEditMode } from './geojson-edit-mode';
 
-const DEFAULT_TOOLTIPS = [];
-
 export class MeasureDistanceMode extends GeoJsonEditMode {
-  startingPoint: Readonly<EditHandleFeature> | null | undefined = null;
-  endingPoint: Readonly<EditHandleFeature> | null | undefined = null;
-  endingPointLocked = false;
+  _isMeasuringSessionFinished = false;
+  _currentTooltips = [];
+  _currentDistance = 0;
 
-  _setEndingPoint(mapCoords: Position) {
-    this.endingPoint = {
-      type: 'Feature',
-      properties: {
-        guideType: 'editHandle',
-        editHandleType: 'existing',
-        featureIndex: -1,
-        positionIndexes: [],
-      },
-      geometry: {
-        type: 'Point',
-        coordinates: mapCoords,
-      },
-    };
-  }
+  _calculateDistanceForTooltip = memoize(({ positionA, positionB, modeConfig }) => {
```

Comment:
Since you are calling this with different arguments (every time the pointer moves, it is different coordinates), I don't think memoizing is actually buying you anything here. Looks like the memoize implementation we have only caches that args of the last call.

Reply:
I agree. Will fix it